### ISSUE TITLE
Disabling file.js module

### ISSUE THREAD

**Issue by wehriam** [amark/gun#401]
Cannot disable the default file.js module when using another storage adapter in Node.js.

## Expected Behavior

The file.js module should not load.

## Current Behavior

The file.js loads and creates a data.json file.

```
WARNING! This `file.js` module for gun is intended for local development testing only!
```

## Possible Solution

Update https://github.com/amark/gun/blob/master/lib/file.js#L13-L17 to test for a boolean false and return.

```js
if(opt.file === false) {
  return;
}
```

## Steps to Reproduce (for bugs)

```js

const levelup = require('levelup');
const leveldown = require('leveldown');
const Gun = require('gun');
require('gun-level');

const levelDB = levelup('data', {
  db: leveldown,
});

const gun = Gun({
  level: levelDB,
  file: false,
});

```

## Your Environment

* Version used: 0.8.3
* Node.js version: v8.2.1
* Operating System: OSX 10.12.6
* Link to your project: N/A

**Comment by amark** [comment]
Hey, sorry... how to do this isn't document. I apologize. (We were just chatting about this on another PR yesterday and in the gitter channel, and on a couple of the other storage adapters).

Your possible solution is what we were using/doing for a while. :) So spot on! And thanks a ton for jumping in and participating and making a PR! That is an excellent way to contribute. Thanks so much!

What we wound up doing instead, is making the `file` option generic for other storage adapters to use (if they want). And instead, the way you disable `file.js` module is you simply don't include by doing:

```
const levelup = require('levelup');
const leveldown = require('leveldown');
const Gun = require('gun/gun'); // this includes core, no defaults!
require('gun-level');
```

## What is going on here?

In NodeJS `require('gun')` references `index.js` which points to `lib/server.js` which is a simple file that automatically requires some default adapters that are useful for NodeJS / server side. ( https://github.com/amark/gun/blob/master/lib/server.js ).

One of the things it includes by default is `lib/ws` which is the websocket server. So, to make  a more complete example of what you might want, you'd probably do something like:

```
const levelup = require('levelup');
const leveldown = require('leveldown');
const Gun = require('gun/gun'); // this includes core, no defaults!
require('gun/lib/server');
require('gun/nts');
require('gun-level');
```

Basically, you pick and choose which modules you want.

I hope this helps! And hopefully we'll get it documented - sorry that we haven't, there was no way for you to know!

Again, I super super appreciate you jumping in and making the PR to help out. That is stellar!

**Comment by wehriam** [comment]
Thanks. How would I specify a custom http server?

**Comment by amark** [comment]
Oops, I updated my previous comment.

`require('gun/lib/ws')` is the default websocket server that connects with your http server.

Does that answer your question?

**Comment by wehriam** [comment]
Previously to use a custom http server it would be something like:

```js
const gun = Gun({
    web: server,
});
```

Is there an equivalent option with this new setup mechanism?

**Comment by amark** [comment]
Yes, that will work the same if you include `require('gun/lib/ws')` yes :) (or even if you included `require('gun/lib/uws')` but you'd need to first install `npm install uws`).

Again, all `require('gun')` (versus gun/gun) does is point to `lib/server.js` which is a 7 line file that pulls in default adapters (like `lib/file.js`, `lib/ws.js`, and such).

Let me know if that works for you, and/or you need any other help. (Again, check out lib/server.js as a reference point).

**Comment by amark** [comment]
I wound up accepting the change but under a different name, you can do this server-side with `Gun({localStorage: false})` because the `file.js` module is based on the browser's localStorage module. The `file` option is reserved for a generic option to change the name of the file/folder a storage engine saves to. But yes, you can now disable the file.js with just an option now. Closing!

**Comment by lhr0909** [comment]
@amark I have found this warning showing up when I tried to use `localStorage: false` on server side, and a `radata` folder appears. Is this intended?

https://github.com/amark/gun/blob/master/lib/store.js#L11

I can send you a PR for this if you let me know what to do. Cheers.

**Comment by amark** [comment]
@lhr0909 this is a good point, you are right, this is probably confusing.

Here is what is going on: (maybe you have some ideas/suggestions?)

1. file.js (roughly the same as the localStorage code, except in NodeJS) is the default storage adapter. It is not recommended to run in production.

2. We're in the middle of building a scalable storage engine, the Radix Storage engine (related to the code you linked to). It will be the new default once we finish testing it.

3. If you set `localStorage: false` that will disable it in the browser (if that option is set in the browser) and in NodeJS. However, rather than defaulting to 0 storage, I made it "switch" to the currently-in-testing Radix Storage Engine instead, which people might not expect.

This is what I'll probably do?

A) Add an option `radisk: false` which will disable the Radix Storage Engine.

That way if you want to turn BOTH off `localStorage: false, radisk: false` could be set.

In the future, the file.js module won't be included by default anyways (so there will be no need to set localStorage: false in NodeJS), as the Radix Storage Engine will be the new default. I'll then provide a separate easy-option to disable it.

Sound good?
